kie-server-parent/kie-server-remote/kie-server-rest/kie-server-rest-jbpm/src/main/java/org/kie/server/remote/rest/jbpm/ProcessResource.java

```diff
@@ -668,18 +677,26 @@ public Response getProcessesByDeploymentId(@Context HttpHeaders headers,
             @ApiParam(value = "optional sort column, no default", required = false) @QueryParam("sort") String sort, 
             @ApiParam(value = "optional sort direction (asc, desc) - defaults to asc", required = false) @QueryParam("sortOrder") @DefaultValue("true") boolean sortOrder) {
 
+        Variant v = getVariant(headers);
         Header conversationIdHeader = buildConversationIdHeader(containerId, context, headers);
-
-        ProcessDefinitionList processDefinitionList = runtimeDataServiceBase.getProcessesByDeploymentId(containerId, page, pageSize, sort, sortOrder);
-        logger.debug("Returning result of process definition search: {}", processDefinitionList);
-
-        return createCorrectVariant(processDefinitionList, headers, Response.Status.OK, conversationIdHeader);
-
+        try {
+            ProcessDefinitionList processDefinitionList = runtimeDataServiceBase.getProcessesByDeploymentId(containerId, page, pageSize, sort, sortOrder);
+            logger.debug("Returning result of process definition search: {}", processDefinitionList);
+    
+            return createCorrectVariant(processDefinitionList, headers, Response.Status.OK, conversationIdHeader);
+            
+        } catch (DeploymentNotFoundException e) {
+            return notFound(MessageFormat.format(CONTAINER_NOT_FOUND, containerId), v, conversationIdHeader);
+        } catch (Exception e) {
+            logger.error("Unexpected error during processing {}", e.getMessage(), e);
+            return internalServerError(MessageFormat.format(UNEXPECTED_ERROR, e.getMessage()), v, conversationIdHeader);
+        }
     }
 
     @ApiOperation(value="Retrieves node instances for given process instance. Depending on provided query parameters (activeOnly or completedOnly) will return active and/or completes nodes",
             response=NodeInstanceList.class, code=200)
-    @ApiResponses(value = { @ApiResponse(code = 500, message = "Unexpected error")})
+    @ApiResponses(value = { @ApiResponse(code = 500, message = "Unexpected error"),
+                            @ApiResponse(code = 404, message = "Process or Container Id not found")})
```

Comment:
could you please change the message to Process **Instance** or container id not found